File: 60-permutation-sequence/60-permutation-sequence.java
class Solution {
    public String getPermutation(int n, int k) {
        List<Integer>temp=new ArrayList<>();
        int fact=1;
        for(int i=1;i<=n;i++){
            temp.add(i);
            if(i==n)continue;
            fact*=i;
        }
        k--;
        String res="";
        while(true){
            
            int t=k/fact;
            res+=temp.get(t);
            
            temp.remove(t);
            if(temp.size()==0)break;
            k=k%fact;
            fact=fact/temp.size();
            
        }
        return res;
        
    }
}

File: 60-permutation-sequence/60-permutation-sequence.cpp
class Solution {
public:
    string getPermutation(int n, int k) {
        string res;
        vector<int>num;
        int fact=1;
        for(int i=1;i<=n;i++){
            num.push_back(i);
            if(i==n)break;
            fact*=i;
        }
        k--;
        while(n){
           res+=to_string(num[k/fact]);
            int rem=k%fact;
            num.erase(num.begin()+k/fact);
            n--;
            if(n==0)return res;
            fact/=n;
            k=rem;
            
            
        }
        return res;
        
        
    }
};